src/pushsource/_impl/model/ami.py

```diff
@@ -107,6 +160,18 @@ class AmiPushItem(VMIPushItem):
     )
     """``True`` if the image is allowed to be released publicly (shared with group "all")."""
 
+    scanning_port = attr.ib(type=int, default=0, validator=optional(instance_of(int)))
+    """AMI scanning port, used when importing the AMI into AWS Marketplace to validate the AMI."""
+
+    user_name = attr.ib(type=str, default=None, validator=optional_str)
+    """The username used to login to the AMI. (Usually set to ec2-user)"""
+
+    version_title = attr.ib(type=str, default=None, validator=optional_str)
+    """The title given to a version. This will display in AWS Marketplace as the name of the version."""
+
+    security_groups = attr.ib(type=list, default=attr.Factory(frozenlist))
```

Comment:
Please ensure there's `converter=frozenlist` as well so that the object remains immutable.